src/common/sourceUtils.ts

```diff
@@ -328,8 +325,10 @@ export function getOptimalCompiledPosition(
   allLocations.push([prevLocation as INotNullRange, prevVariance]);
 
   // Sort again--sort is stable (de facto for a while, formalized in ECMA 2019),
-  // so we get the first location that has the least variance.
-  allLocations.sort(([, varA], [, varB]) => varA - varB);
+  // so we get the first location that has the least variance, or if the variance is the same, the one that appears earliest.
+  allLocations.sort(
+    ([a, varA], [b, varB]) => varA - varB || a.line - b.line || a.column - b.column,
```

Comment:
technically you only need `varA - varB` since we sorted by location earlier and .sort() is stable